master/buildbot/test/fake/fakedb.py

```diff
@@ -90,6 +91,10 @@ def __init__(self, **kwargs):
         for k, v in iteritems(self.values):
```

Comment:
I don't like this part. If all strings should be unicode in the interface IMO it's better to add assert that they are, e.g.:

```
for k, v in iteritems(self.values):
    assert not isinstance(v, str), "All strings passed to DB layer must be unicode "
```

and fix tests to pass only unicode strings.